website/docs/r/private_key.html.md

```diff
@@ -24,43 +23,71 @@ state and does not create any external managed resources.
 ## Example Usage
 
 ```hcl
+// ECDSA key with P384 elliptic curve
 resource "tls_private_key" "example" {
   algorithm   = "ECDSA"
   ecdsa_curve = "P384"
 }
+
+// RSA key of size 4096 bits
+resource "tls_private_key" "example" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+// ED25519 key
+resource "tls_private_key" "example" {
+  algorithm = "ED25519"
+}
 ```
 
 ## Argument Reference
 
 The following arguments are supported:
 
-* `algorithm` - (Required) The name of the algorithm to use for
-the key. Currently-supported values are "RSA" and "ECDSA".
+* `algorithm` - (Required) The name of the algorithm to use when generating
+the key. Currently-supported values are `RSA`, `ECDSA` and `ED25519`.
```

Comment:
The algorithm names can be enforced at validation-time by using the [`helper/validation.StringInSlice` function](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation#StringInSlice) on the schema attribute, e.g.

```go
			"algorithm": {
				Type:        schema.TypeString,
				Required:    true,
				Description: "Name of the algorithm to use to generate the private key",
				ForceNew:    true,
				ValidateFunc: validation.StringInSlice([]string{ECDSA, ED25519, RSA}),
			},
```

Reply:
Oh, this is super nice!
Yes, absolutely!